### ISSUE TITLE
proxy not found when doing stage out on condor_g

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRAB2#729]
Original Savannah ticket 87233 reported by belforte on Wed Sep 28 08:08:45 2011.

latest tests on condor_g with 279pre4 indicates that
stage out command does not find the user's proxy.
see e.g.
https://hypernews.cern.ch/HyperNews/CMS/get/crabDevelopment/1541/1/2.html

exact reason to be found

**Comment by ericvaandering** [comment]
Comment by belforte on Wed Sep 28 09:49:55 2011

proxy is transported allright, seems what happens is that lcg-cp can not find it, in spite of X509_USER_PROXY being defined
adding some commands at top of crab wrapper I get:

&gt;&gt;&gt; voms proxy information:
subject   : /C=IT/O=INFN/OU=Personal Certificate/L=Trieste/CN=Stefano Belforte/CN=proxy/CN=limited proxy
issuer    : /C=IT/O=INFN/OU=Personal Certificate/L=Trieste/CN=Stefano Belforte/CN=proxy
identity  : /C=IT/O=INFN/OU=Personal Certificate/L=Trieste/CN=Stefano Belforte/CN=proxy
type      : limited proxy
strength  : 1024 bits
path      : /var/condor/.execute-links/slot09/dir_3013/x509_up
timeleft  : 191:33:52

X509_USER_PROXY=/var/condor/.execute-links/slot09/dir_3013/x509_up

but later lcg-ls says
which lcg-ls
/afs/hep.wisc.edu/osg/wnclient/lcg/bin/lcg-ls
########### details of SE interaction
2011-09-28 09:08:07.764049:
Executed:       unset LD_LIBRARY_PATH; unset GLITE_ENV_SET; export PATH=/usr/bin:/bin; source /etc/profile; source /scratch/condor/execute/dir_3013/CacheEnv.sh ; lcg-ls -b -D srmv2 --srm-timeout 2400 --sendreceive-timeout 2400 --connect-timeout 300 --verbose srm://cmsdcache.pi.infn.it:8443/srm/managerv2?SFN=/pnfs/pi.infn.it/data/cms/store/user/belforte/testAug24-01/kk_1_1_KVJ.root
Done with exit code:    1
and output:
SE type: SRMv2
[SE][Ls][] httpg://cmsdcache.pi.infn.it:8443/srm/managerv2: CGSI-gSOAP: Could NOT load client credentials
GSS Major Status: General failure

GSS Minor Status Error Chain:
acquire_cred.c:gss_acquire_cred:143:
Error with GSI credential
globus_i_gsi_gss_utils.c:globus_i_gsi_gss_cred_read:1342:
Error with gss credential handle
globus_gsi_credential.c:globus_gsi_cred_read:732:
Valid credentials could not be found in any of the possible locations specified by the credential search order.
Valid credentials could not be found in any of the possible locations specified by the credential search order.

puzzling...

**Comment by ericvaandering** [comment]
Comment by belforte on Thu Sep 29 04:37:00 2011

bug found. The proxy is there allright, but when
submission goes via condor_g it is located in the
current work directory and
we introduced chmod a+w of the directory to fix
https://savannah.cern.ch/bugs/?78585
we ended up in changing also the proxy permission
at which pont globus does not like it any more.
Adding a chmod 600 $X509_USER_PROXY after the
chmod a+w fixed this.

Tests are succesful now.
Fix committed to CVS
/cvs_server/repositories/CMSSW/COMP/CRAB/python/cms_cmssw.py,v  &lt;--  cms_cmssw.py
new revision: 1.379; previous revision: 1.378
